src/constraint.rs

```diff
@@ -109,6 +111,10 @@ impl Constraint {
     ///   In this instance [`Constraint::Compatible`] is a case-insensitive equality comparison.
     pub fn compare(&self, fetcher: Fetcher, target: &Revision) -> bool {
         match fetcher {
+            Fetcher::Gem => gem::compare(self, Fetcher::Gem, target).unwrap_or_else(|err| {
+                warn!("could not parse rubygems version '{err:?}'");
```

Comment:
[nit] Prefer to include the error and associated metadata as structured output:
```suggestion
                warn!(?err, "could not compare version");
```

Ideally this is combined with a `#[tracing::instrument]` on the `compare` method so that we know what is being compared; I just forgot to add this but feel free to do so.